include/mbedtls/asn1.h

```diff
@@ -97,14 +97,13 @@
 /* Slightly smaller way to check if tag is a string tag
  * compared to canonical implementation. */
 #define MBEDTLS_ASN1_IS_STRING_TAG(tag)                                     \
-    ((tag) < 32u && (                                                      \
-         ((1u << (tag)) & ((1u << MBEDTLS_ASN1_BMP_STRING)       |     \
-                           (1u << MBEDTLS_ASN1_UTF8_STRING)      |     \
-                           (1u << MBEDTLS_ASN1_T61_STRING)       |     \
-                           (1u << MBEDTLS_ASN1_IA5_STRING)       |     \
-                           (1u << MBEDTLS_ASN1_UNIVERSAL_STRING) |     \
-                           (1u << MBEDTLS_ASN1_PRINTABLE_STRING) |     \
-                           (1u << MBEDTLS_ASN1_BIT_STRING))) != 0))
+    ((tag) < 32 && (                                                      \
```

Comment:
I really don't like this change, since if `tag` has `-1` this test will now pass and we'll attempt to shift left by `-1`.

Instead, I think a better fix might be to cast `tag` to unsigned for the comparison against `32u`.

Also, the changes from `1u` to `1` aren't necessary, and could be dangerous: `(signed) 1 << 31` is UB, IIRC

Reply:
and (minor): if we're modifying this, let's have all the `\`s line up :)